src/components/Portal/Portal.js

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useRef, useState } from 'react'
+import { createPortal } from 'react-dom'
+import PropTypes from 'prop-types'
+
+/**
+ * Hooks handles the creation and tear down of the portal's root
+ * element.
+ *
+ * @private
+ *
+ * @param {String} id - The id of the target container
```

Comment:
Should maybe mention it's optional and that if not supplied a generic div will be created